internal/tiltfile/tiltfile.go

```diff
@@ -166,6 +167,10 @@ func (tfl tiltfileLoader) Load(ctx context.Context, filename string, matching ma
 
 	yamlManifest := model.Manifest{}
 	if len(unresourced) > 0 {
+		// We apply Namespaces first b/c subsequent entities may depend on the NS existing.
+		// We apply CRDs first so we don't try to create an instance of that CRD before it's defined.
+		sort.Sort(k8s.NamespacesAndCRDsFirst(unresourced))
```

Comment:
would it make sense to do this in NewK8sTarget, so that all K8sTargets get this treatment? i'm pretty sure you can write a Tiltfile that puts the Namespace/CRD in a named manifest

Reply:
alternatively, you could do the Sort in Upsert, which is the only place this really matters

Reply:
Yeah, `Upsert` is probably the right place to do this. IIUC, `Upsert` moves all `Job`s and `Pod`s in front of everything else, so any sorting we do prior to that won't help us if we have both a pod and a namespace.

Reply:
(which was also a bug in the previous solution to this problem!)

Reply:
oh womp. will put this in upsert, good tip!

Reply:
blurgh because of the way we upsert, sorting does us no good -- we pluck stuff out of the entities list to apply. fixing now.